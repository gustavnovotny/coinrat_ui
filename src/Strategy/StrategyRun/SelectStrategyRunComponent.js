// @flow
import React from "react"
import {SelectComponent, SelectItemsType, SelectElement, SORTING_DESC} from "../../Form/Select/SelectComponent"

type Props = {
  availableStrategyRuns: SelectItemsType,
  onSelect: (strategyRunId: string) => void,
  defaultSelectedStrategyRun: string,
  isLoading: boolean
}

const SelectStrategyRunComponent = ({availableStrategyRuns, defaultSelectedStrategyRun, onSelect, isLoading}: Props) => {
  const item = availableStrategyRuns[defaultSelectedStrategyRun]
  return <SelectComponent
    label="Strategy Run"
    items={availableStrategyRuns}
    selectedItem={item || null}
    isLoading={isLoading}
    onChange={(strategyRun: SelectElement) => onSelect(strategyRun.key)}
    sort={SORTING_DESC}
  />
}

export default SelectStrategyRunComponent
