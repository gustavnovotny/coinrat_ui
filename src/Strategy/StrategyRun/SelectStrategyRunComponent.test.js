// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import SelectStrategyRunComponent from "./SelectStrategyRunComponent"

it('renders correctly', () => {
  const onSelect = () => undefined
  const tree = renderer
    .create(
      <SelectStrategyRunComponent
        availableStrategyRuns={{foo: {key: 'foo', title: 'Foo'}}}
        defaultSelectedStrategyRun="foo"
        onSelect={onSelect}
      />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
