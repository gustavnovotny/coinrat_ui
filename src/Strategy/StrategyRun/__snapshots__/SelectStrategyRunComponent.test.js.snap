// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="_Rfx0_"
>
  <div
    className="form-item form-label _Rfx1_"
  >
    <div
      className="pt-label"
    >
      Strategy Run:
      <span
        className="pt-text-muted"
      />
    </div>
  </div>
  <div
    className="form-item form-element"
  >
    <span>
      <div
        className="pt-popover-target"
        onClick={[Function]}
      >
        <div
          className=""
          disabled={undefined}
          onKeyDown={[Function]}
          onKeyUp={undefined}
          tabIndex={undefined}
        >
          <button
            className="pt-button"
            disabled={undefined}
            onClick={undefined}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <div
              className="_Rfx2_ _Rfx0_"
            >
              <div
                className=""
                style={
                  Object {
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Foo
              </div>
              <div
                className=""
              >
                <span
                  className="pt-icon-standard pt-icon-caret-down"
                />
              </div>
            </div>
          </button>
        </div>
      </div>
    </span>
  </div>
  <div
    className="form-item form-suffix"
  >
    
     
    
  </div>
</div>
`;
